stl/inc/memory

```diff
@@ -283,7 +283,7 @@ namespace ranges {
             auto _OFirst      = _Get_unwrapped(_STD move(_First2));
             const auto _OLast = _Get_unwrapped(_STD move(_Last2));
             if constexpr (_Ptr_move_cat<_It, _Out>::_Really_trivial) {
-                _OFirst = _Copy_memcpy_common(_IFirst, _IFirst + _Count, _OFirst, _OLast);
+                return _Copy_memcpy_common(_IFirst, _IFirst + _Count, _OFirst, _OLast);

```

Comment:
Am I confused, or does this assumes that for all iterator types that unwrap to pointer types, pointers values of those types can be implicitly converted to said iterator types? I think we need to rewrap the return value of `_Copy_memcpy_common` before we can return it from `operator()`.

Reply:
I first went with span only to realize that we made span iterator an aggregate so that would work...

Reply:
> I first went with span only to realize that we made span iterator an aggregate so that would work...

Heh - that was my first thought as well. "Use `span` instead of an array. Oh wait..."